app/uk/gov/hmrc/cgtpropertydisposals/service/TaxEnrolmentService.scala

```diff
@@ -16,62 +16,74 @@
+import uk.gov.hmrc.cgtpropertydisposals.actors.TaxEnrolmentRetryManager.AttemptTaxEnrolmentAllocationToGroup
+import uk.gov.hmrc.cgtpropertydisposals.models.{Error, TaxEnrolmentRequest}
+import uk.gov.hmrc.cgtpropertydisposals.modules.TaxEnrolmentRetryProvider
+import uk.gov.hmrc.cgtpropertydisposals.service.TaxEnrolmentService.TaxEnrolmentDatabaseInsertResponse
+import uk.gov.hmrc.cgtpropertydisposals.util.Logging
 import uk.gov.hmrc.http.HeaderCarrier
 
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
-
+import scala.concurrent.duration._
 @ImplementedBy(classOf[TaxEnrolmentServiceImpl])
 trait TaxEnrolmentService {
-  def allocateEnrolmentToGroup(cgtReference: String, subscriptionDetails: SubscriptionDetails)(
+  def allocateEnrolmentToGroup(taxEnrolmentRequest: TaxEnrolmentRequest)(
     implicit hc: HeaderCarrier
-  ): EitherT[Future, Error, TaxEnrolmentResponse]
-}
-
-object TaxEnrolmentService {
-
-  sealed trait TaxEnrolmentResponse
-
-  object TaxEnrolmentResponse {
-    case object TaxEnrolmentCreated extends TaxEnrolmentResponse
-  }
-
+  ): EitherT[Future, Error, Unit]
 }
 
 @Singleton
-class TaxEnrolmentServiceImpl @Inject()(connector: TaxEnrolmentConnector) extends TaxEnrolmentService {
+class TaxEnrolmentServiceImpl @Inject()(
+  taxEnrolmentRetryProvider: TaxEnrolmentRetryProvider
+) extends TaxEnrolmentService
+    with Logging {
 
-  override def allocateEnrolmentToGroup(cgtReference: String, subscriptionDetails: SubscriptionDetails)(
+  override def allocateEnrolmentToGroup(taxEnrolmentRequest: TaxEnrolmentRequest)(
     implicit hc: HeaderCarrier
-  ): EitherT[Future, Error, TaxEnrolmentResponse] = {
-
-    val enrolmentRequest = subscriptionDetails.address match {
-      case Address.UkAddress(line1, line2, line3, line4, postcode) =>
-        EnrolmentRequest(List(KeyValuePair("Postcode", postcode)), List(KeyValuePair("CGTPDRef", cgtReference)))
-      case Address.NonUkAddress(line1, line2, line3, line4, maybePostcode, countryCode) =>
-        EnrolmentRequest(List(KeyValuePair("CountryCode", countryCode.code)), List(KeyValuePair("CGTPDRef", cgtReference)))
-    }
+  ): EitherT[Future, Error, Unit] = {
+    implicit val timeout: Timeout = Timeout(5.seconds)
 
-    connector
-      .allocateEnrolmentToGroup(cgtReference, enrolmentRequest)
-      .subflatMap { response =>
-        response.status match {
-          case NO_CONTENT =>
-            Right(TaxEnrolmentCreated)
-          case UNAUTHORIZED =>
-            Left(Error("Unauthorized"))
-          case BAD_REQUEST =>
-            Left(Error("Bad request"))
-          case other => Left(Error(s"Received unexpected http status in response from tax-enrolment service"))
+    EitherT[Future, Error, Unit](
+      taxEnrolmentRetryProvider.taxEnrolmentRetryManager
+        .ask(AttemptTaxEnrolmentAllocationToGroup(taxEnrolmentRequest))
+        .mapTo[TaxEnrolmentDatabaseInsertResponse]
```

Comment:
`TaxEnrolmentDatabaseInsertResponse` as a response to `AttemptTaxEnrolmentAllocationToGroup` sounds a bit funny - I'd have expecting something like `AttemptTaxEnrolmentAllocationToGroupResponse` . Unless the request message should be `TaxEnrolmentDatabaseInsertRequest`. It just sounds like the request and response are for two different things

Reply:
the database errors could be handled in the actor level and a generic error could be surfaced in the response - that way  there's less changes to make if we ever add another type of error which should result in failure